### ISSUE TITLE
Javascript client for using Tigris

### ISSUE THREAD

**Issue by adilansari** [tigrisdata/tigris#1048]
**Is your feature request related to a problem? Please describe.**
I would like to use Tigris in my front end application. The existing TypeScript client library is designed specifically for server side use cases.

**Describe the solution you'd like**
A JS client library

**Describe alternatives you've considered**
Using TypeScript client library

**Comment by garrensmith** [comment]
Typescript isn't blocking us from running on the client. Most frontend frameworks have typescript types and are written in typescript. What is stopping us is the grpc in our current client. I think we could solve this by making our current client repo into a monorepo. And then taking the current grpc part and making that a specific network layer. We had another layer that uses fetch/http that we can pass through to the client. We then would create two packages we release. The tigris-grpc version and the tigris HTTP version.

**Comment by ovaistariq** [comment]
@garrensmith, you are right. @adilansari and I chatted about it and we mainly need a client that uses fetch API instead of grpc. @adilansari will be closing this and creating an issue on the tigris-client-ts repo.

**Comment by himank** [comment]
+1 it is mainly grpc. This is an excellent idea for separating the network layer.

**Comment by garrensmith** [comment]
One final thing, this is a common issue and the idea of a separate network layer is what we did with PouchDB and Prisma.
